frontend/src/board/board.test.ts

```diff
@@ -46,3 +47,93 @@ test(
     ])
   },
 );
+
+
+test('Checking eligible moves in the North direction', () => {
+  let board = new Board();
+  expect(board.checkDirections(6, 15, 'North')).toStrictEqual(
+    { row: 0, column: 15 })
+}
+)
+test('Checking eligible moves in the South direction', () => {
+  let board = new Board();
+  expect(board.checkDirections(6, 15, 'South')).toStrictEqual(
+    { row: 15, column: 15 })
+}
+)
+
+test('Checking eligible moves in the East direction', () => {
+  let board = new Board();
+  expect(board.checkDirections(6, 14, 'East')).toStrictEqual(
+    {row: 6, column: 15})
+}
+)
+test('Checking eligible move in the West direction', () => {
+  let board = new Board();
+  expect(board.checkDirections(6, 15, 'West')).toStrictEqual(
+  {row: 6, column: 0})
+}
+)
+
+test("Should return target cell in path", () => {
+  let board = new Board();
+  if (board.cells[15] && board.cells[15][14]) {
+    board.cells[15][14].isTarget = true;
+  }
+  expect(board.checkDirections(9,14,"South")).toStrictEqual({row:15,column:14})
+  expect(board.checkDirections(15,9,"East")).toStrictEqual({row:15,column:14})
```

Comment:
See comments in `board.ts` but I don't think this is actually the behavior we want - the robot shouldn't stop sliding unless it runs into something